php/pantheon/checks/plugins.php

```diff
@@ -37,42 +37,52 @@ public function run() {
 		$all_plugins = Utils::sanitize_data( get_plugins() );
 		$update = Utils::sanitize_data( get_plugin_updates() );
 		$report = array();
+		$should_check_vulnerabilities = $this->getWpScanApiToken();
+		$vulnerable = false;
+
 		foreach( $all_plugins as $plugin_path => $data ) {
 			$slug = $plugin_path;
 			if (stripos($plugin_path,'/')) {
 				$slug = substr($plugin_path, 0, stripos($plugin_path,'/'));
 			}
 
-			$vulnerable = $this->is_vulnerable($slug, $data['Version']);
+			if ( $should_check_vulnerabilities ) {
+				$vulnerable = $this->is_vulnerable($slug, $data['Version']);
+			}
 
 			$needs_update = 0;
 			$available = '-';
 			if (isset($update[$plugin_path])) {
 				$needs_update = 1;
 				$available = $update[$plugin_path]->update->new_version;
 			}
-			if ( false === $vulnerable ) {
-				$vulnerable = "None";
-			} else {
+
+			if ( $should_check_vulnerabilities && isset( $vulnerable ) ) {
```

Comment:
Can this isset fail at all? Unless line 50 deletes the variable entirely, I would think it could not.

Reply:
[PHP.net describes `isset`](http://www.php.net/manual/en/function.isset.php) as "isset — Determine if a variable is declared and is different than `null`"

`$vulnerable` is initially set to `false` and then updated to the result of `$this->is_vulnerable` which returns `false` unless there is a vulnerability and only if we `$should_check_vulnerabilities` -- so, still `false`. `false !== null` so I assume this should validate with the intended purpose but I suppose we could leave it the way it was originally and explicitly check against `false`, or even if it's truthy at all. I based this check on the assumption that using `isset` to distinguish `false` as different from `null` was sufficient.